eng/tools/spec-gen-sdk-runner/src/commands.ts

```diff
@@ -187,29 +195,52 @@ export async function generateSdkForBatchSpecs(batchType: string): Promise<numbe
   // Construct the spec-gen-sdk command
   const specGenSdkCommand = prepareSpecGenSdkCommand(commandInput);
   // Get the spec paths based on the batch run type
-  const specConfigPaths = getSpecPaths(batchType, commandInput.localSpecRepoPath);
+  const specConfigsArray: SpecConfigs[] = getSpecPaths(batchType, commandInput.localSpecRepoPath);
 
   // Prepare variables
   let statusCode = 0;
+  let pushedSpecConfigCount;
   let markdownContent = "\n";
   markdownContent += `## Batch Run Type\n ${batchType}\n`;
   let failedContent = `## Spec Failures in the Generation Process\n`;
   let succeededContent = `## Successful Specs in the Generation Process\n`;
   let notEnabledContent = `## Specs with SDK Not Enabled\n`;
+  let duplicatedConfigContent = `## Specs with Duplicated SDK Configurations (in 'tspconfig.yaml' and 'readme.md')\n`;
   let failedCount = 0;
   let notEnabledCount = 0;
+  let duplicatedConfigCount = 0;
   let succeededCount = 0;
   let executionReport;
+  let specConfigPath = "";
+  let stagedArtifactsFolder = "";
 
   // Generate SDKs for each spec
-  for (const specConfigPath of specConfigPaths) {
-    logMessage(`Generating SDK from ${specConfigPath}`, LogLevel.Group);
+  for (const specConfigs of specConfigsArray) {
+    if (specConfigs.tspconfigPath && specConfigs.readmePath) {
+      logMessage(
+        `Generating SDK from ${specConfigs.tspconfigPath} and ${specConfigs.readmePath}`,
```

Comment:
Why are we generating from both? I would expect us to just warn and use the tsp configuration.

Reply:
Or are we just passing both not necessarily generating both as I suspect spec-gen-sdk is only generating from tsp.

Reply:
We only generate from TypeSpec if both configs exist and report the warnings of duplicated config issue. Here is just the logging to differentiate between one input config and two input configs.
![image](https://github.com/user-attachments/assets/2f3827b4-7707-4372-90cc-e790e7f06e4c)

Reply:
OK I think this log message might cause confusion and we should either remove it and leave it up to spec-gen-sdk tool to log and in the runner we just pass whichever parameters we have.

Reply:
I knew that it's a little confusion, but I haven't come with a better solution. The group log is written by runner, and runner passes spec configs to spec-gen-sdk tool. Then spec-gen-sdk tool validate spec configs and determine the right configuration to be used for code generation. Currently users are able to know the details when expanding the logs.
What do you suggest for the group log title?

Reply:
I would just remove the and part. In fact, just remove this entire if block and let the below if/else write generating for tsp (if tsp is set ignoring the readme) and then generating from the readme, if only the readme is present.

Reply:
@weshaggard I updated it to log the service folder as group title. It's not perfect but it should not be fine for now?

Reply:
Let's start with that as it should at least help with some confusion.